AutoRefAPI.md

```diff
@@ -0,0 +1,187 @@
+# AutoReferee Communication Interface
+
+Instead of using the visual interface, the GameController can also be
+operated using an automatic referee system (AutoRef). 
+
+If you face any problems with the implementation or API, please 
+create an Issue on github to report it.
+
+## Setup
+
+
+### 1. Building from Source
+
+First, you need to build the GameController from source by calling `ant`
+in the main directory. For more details, see the official README.md.
+
+
+### 2. Requirements
+
+The GameController expects a `game.json` file to be present in 
+`resources/config/sim`. The config file needs to contain:
+
+````json
+{
+  "type": "NORMAL" | "KNOCKOUT",
+  "class": "ADULT" | "KID",
+  "blue": {
+    "id": <team_id>
+  },
+  "red":{
+    "id": <team_id>
+  }
+}
+````
+The `team_id` is the unique identifier a team is given in the beginning
+of the tournament. It needs to be an identifier that is set in the 
+`teams.cfg` file in the respective `hl_sim_adult` or `hl_sim_kid` config
+folder.
+
+
+
+### 3. Executing the Jar
+
+To run the GameController with an instances that can be accessed from the
+AutoRef, you need to execute the `GameControllerSimulator.jar`:
+
+```bash
+cd build/jar
+java -jar GameControllerSimulator.jar
+```
+
+### 4. Testing
+
+To test the TCP/IP message interface, you can run the ``main.py`` script
+in ``sim_client`` using Python 3.
+
+## TCP/IP Connection
+
+To connect to the GameController via the AutoReferee, you need to
+open a TCP/IP connection on port 8750. The port can be configured in
+the `AutoRefServer.java` class. 
```

Comment:
Maybe add this to the config file.

Reply:
I actually don't think this makes sense in the config file but I will add a launch parameter for this case